spring-core/src/main/java/org/springframework/util/function/SingletonSupplier.java

```diff
@@ -41,6 +42,15 @@
  */
 public class SingletonSupplier<T> implements Supplier<@Nullable T> {
 
+	private static final Object NULL = new Object() {
+
+		@Override
+		public String toString() {
```

Comment:
It seems that overriding this method is not necessary, what's the purpose?

Reply:
The reason that the method is overridden isn't for functionality but for debugging purposes.

<img width="315" height="25" alt="Screenshot 2025-08-25 at 10 08 17" src="https://github.com/user-attachments/assets/27d2ff98-9f8e-4876-b981-504dd34600d8" /> 

When overridden:
<img width="443" height="29" alt="Screenshot 2025-08-25 at 10 07 57" src="https://github.com/user-attachments/assets/b8ce5484-3a66-4ebb-9a76-3d16c662faa6" />